controllers/machinehealthcheck_controller_test.go

```diff
@@ -1460,3 +1382,232 @@ func ownerReferenceForCluster(ctx context.Context, c *clusterv1.Cluster) metav1.
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-cluster-",
+			Namespace:    ns.Name,
+		},
+	}
+	g.Expect(testEnv.Create(ctx, cluster)).To(Succeed())
+	g.Eventually(func() error {
+		var cl clusterv1.Cluster
+		return testEnv.Get(ctx, util.ObjectKey(cluster), &cl)
+	}, timeout, time.Second).Should(Succeed())
+
+	g.Expect(testEnv.CreateKubeconfigSecret(cluster)).To(Succeed())
+	mhc := &clusterv1.MachineHealthCheck{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-mhc-",
+			Namespace:    cluster.Namespace,
+		},
+		Spec: clusterv1.MachineHealthCheckSpec{
+			Selector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"selector": string(uuid.NewUUID()),
+				},
+			},
+			NodeStartupTimeout: &metav1.Duration{Duration: 1 * time.Millisecond},
+			UnhealthyConditions: []clusterv1.UnhealthyCondition{
+				{
+					Type:    corev1.NodeReady,
+					Status:  corev1.ConditionUnknown,
+					Timeout: metav1.Duration{Duration: 5 * time.Minute},
+				},
+			},
+		},
+	}
+	return cluster, mhc.DeepCopy()
+}
+
+func newMachine(mhc *clusterv1.MachineHealthCheck, cluster *clusterv1.Cluster) clusterv1.Machine {
+	return clusterv1.Machine{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: clusterv1.GroupVersion.String(),
+			Kind:       "Machine",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-mhc-machine-",
+			Namespace:    mhc.Namespace,
+			Labels:       mhc.Spec.Selector.MatchLabels,
+		},
+		Spec: clusterv1.MachineSpec{
+			ClusterName: cluster.Name,
+			Bootstrap: clusterv1.Bootstrap{
+				DataSecretName: pointer.StringPtr("data-secret-name"),
+			},
+		},
+		Status: clusterv1.MachineStatus{
+			InfrastructureReady: true,
+			BootstrapReady:      true,
+			Phase:               string(clusterv1.MachinePhaseRunning),
+			ObservedGeneration:  1,
+		},
+	}
+}
+
+func newNode() corev1.Node {
+	node := corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-mhc-node-",
+		},
+		Status: corev1.NodeStatus{
+			Conditions: []corev1.NodeCondition{
+				{Type: corev1.NodeReady, Status: corev1.ConditionTrue},
+			},
+		},
+	}
+	return node
+}
+
+func setNodeUnhealthy(node *corev1.Node) {
+	node.Status.Conditions[0] = corev1.NodeCondition{
+		Type:               corev1.NodeReady,
+		Status:             corev1.ConditionUnknown,
+		LastTransitionTime: metav1.NewTime(time.Now().Add(-10 * time.Minute)),
+	}
+}
+
+func newInfraMachine(machine *clusterv1.Machine) (*unstructured.Unstructured, string) {
+	providerID := fmt.Sprintf("test:////%v", uuid.NewUUID())
+	return &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "infrastructure.cluster.x-k8s.io/v1alpha3",
+			"kind":       "InfrastructureMachine",
+			"metadata": map[string]interface{}{
+				"generateName": "test-mhc-machine-infra-",
+				"namespace":    machine.Namespace,
+			},
+			"spec": map[string]interface{}{
+				"providerID": providerID,
+			},
+		},
+	}, providerID
+}
+
+func fakeNodesMachines(
```

Comment:
second thought on this:
this func implements  a different pattern than the one in newXX or in setup, because in this case we are returning some of the objects and a cleanup func, while in both the other cases we are not returning a cleanup func.
What about dropping the cleanup parameter ad return the infrastructureMachines instead